src/Avalonia.Themes.Fluent/FluentTheme.cs

```diff
@@ -74,6 +92,25 @@ protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T>
                     (Loaded as Styles)![2] = _fluentLight[1];
                 }
             }
+
+            if (change.Property == DensityStyleProperty)
+            {
+                if (DensityStyle == DensityStyle.Compact)
+                {
+                    if ((Loaded as Styles)!.Count > 3)
+                    {
+                        (Loaded as Styles)![3] = _densityStyles[0];
+                    }
```

Comment:
this part of code makes sense only in the case previously DensityStyle had value DensityStyle.Compact I think? We can just ignore changes when oldwalue==newwalue